From 982ab6295580df8c4a3bcb041af25bec723261a2 Mon Sep 17 00:00:00 2001
From: Aloka Dixit <alokad@codeaurora.org>
Date: Wed, 15 Jul 2020 12:45:58 -0700
Subject: [PATCH] hostapd: Fix max BSSID indicator in MBSSID IE

The calculation of max BSSID indicator should include both
transmitting and non-transmitting profiles.
Also, for one non-transmitting VAP, the max BSSID
indicator should be 1 however it is set to 2.
This patch fixes both issues.

Signed-off-by: Aloka Dixit <alokad@codeaurora.org>
---
 src/ap/ieee802_11.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -5730,9 +5730,9 @@ static u8 * hostapd_eid_multiple_bssid_c
 
 multiple_bssid_too_big:
 	*count = i;
-	*num_offset = (u8)ceil(log2(num));
-	if (*num_offset < 2)
-		*num_offset = 2;
+	*num_offset = (u8)ceil(log2(hapd->iface->num_bss));
+	if (*num_offset < 1)
+		*num_offset = 1;
 	*size_offset = (eid - size_offset) - 1;
 
 	return eid;
